src/ai/susi/server/api/aaa/ChangeUserRoles.java

```diff
@@ -149,15 +149,15 @@ public ServiceResponse serviceImpl(Query query, HttpServletResponse response, Au
             ClientCredential credential = new ClientCredential(ClientCredential.Type.passwd_login, userTobeUpgraded);
             ClientIdentity identity = new ClientIdentity(ClientIdentity.Type.email, credential.getName());
             if (!DAO.hasAuthorization(identity)) {
-                throw new APIException(400, "Username not found");
+                throw new APIException(404, "Username not found");
             }
 
             // Update the local database
             Authorization auth = DAO.getAuthorization(identity);
             try {
                 auth.setUserRole(userRole);
             } catch (IllegalArgumentException e) {
-                throw new APIException(400, "role not found");
```

Comment:
@Orbiter just to be sure 

But the exception was thrown because the query string didn't had proper role argument so instead of a `404` it should be a `400` -> Bad Data

What do you suggest about it